lib/debug/tracer.rb

```diff
@@ -156,7 +156,7 @@ def setup
         next if skip?(tp)
 
         if tp.self.object_id == @obj_id
-          out tp, "`#{@obj_inspect}` is used as a receiver of #{minfo(tp)}"
+          out tp, "`#{@obj_inspect}` receives #{minfo(tp)}"
```

Comment:
"obj receives C#foo" doesn't make sense.
"obj receives the message #foo" does make sense. But "C" is gone.

Reply:
ideas:
"C#foo is called with obj as a receiver"
"C#foo is called with obj as a parameter"

Reply:
> C#foo is called with obj as a receiver 

The subject of this sentence is `C#foo` instead of `obj`. But in the context of `trace pass`, I think the `obj` should be highlighted as the subject like the current messages do.

Also, I might be wrong here, but from my understanding `C#foo is called with obj as a receiver` is not grammatically correct either.

`C#foo is called with obj` implies `obj` is an argument passed to it, not the receiver of this call. An example found in `python`'s [unittest.mock document](https://docs.python.org/3/library/unittest.mock.html):

>  The function is called with the same arguments as the mock, and unless it returns DEFAULT

And `as a receiver` still refers to the subject of this sentence, which is the `C#foo`, not the `obj`. 

I also don't think the modified version "not making sense". In general, `being a receiver of something` == `receives something`. For example:

- I'm the receiver of a letter from my friend.
- I received a letter from my friend.

So I think this change still expresses the similar idea.

Reply:
I also don't think we need to pursue 100% grammatical correctness in these messages. As long as it's not confusing with one another, keeping the message short should be prioritized.

Reply:
another idea:

"obj.foo (C#foo)"
"obj.foo" (for singleton method)

Reply:
how about

- `obj receives #foo (C#foo)` for `C#foo`
- `C receives .bar (C.bar)` for `C.bar`
- `obj receives .singleton_foo` for singleton method

Reply:
"obj.foo (C#foo)" doesn't enough?

Reply:
because we use `inspect`'s result, `obj` doesn't always look like an object. for example:
```
`#<ActiveRecord::Relation [#<Post id: 1, title: ...`
```

in such condition, that format isn't easy to read

```
`#<ActiveRecord::Relation [#<Post id: 1, title: ...`#where (ActiveRecord::Relation#where)`
```

so I think this is better

```
`#<ActiveRecord::Relation [#<Post id: 1, title: ...` receives #where (ActiveRecord::Relation#where)`
```

Reply:
I've opened another PR https://github.com/ruby/debug/pull/214 for this conversation and dropped this change because it's not the goal of this PR 😅